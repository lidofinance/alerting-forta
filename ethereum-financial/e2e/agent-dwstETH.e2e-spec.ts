import { configureContainer, Finding } from "forta-agent";
import { AwilixContainer, asFunction } from "awilix";
import {
  provideAgentPath,
  provideRunBlock,
  provideRunTransaction,
} from "./utils";

const TEST_TIMEOUT = 60_000; // ms

describe("agent-dwstETH e2e tests", () => {
  let runBlock: (
    blockHashOrNumber: string | number,
    initBlock?: number,
  ) => Promise<Finding[]>;
  let runTransaction: (txHash: string) => Promise<Finding[]>;
  let logSpy: jest.SpyInstance;
  let timeSpy: jest.SpyInstance;

  beforeAll(() => {
    logSpy = jest.spyOn(console, "log");
    logSpy.mockImplementation(() => {});
    timeSpy = jest.spyOn(Date, "now");
    timeSpy.mockImplementation(() => new Date("2023-12-31"));
  });

  beforeEach(async () => {
    const container = configureContainer() as AwilixContainer;
    container.register({
      agentPath: asFunction(
        provideAgentPath("subagents/dwstETH/agent-dwstETH"),
      ),
      runTransaction: asFunction(provideRunTransaction),
      runBlock: asFunction(provideRunBlock),
    });

    // https://docs.forta.network/en/latest/cli/#invoke-commands-programmatically
    runTransaction = container.resolve("runTransaction");
    runBlock = container.resolve("runBlock");
  });

  afterAll(() => {
    jest.resetAllMocks();
  });

  it(
    "should process block with high amount of dwstETH minted",
    async () => {
      await runTransaction(
        "0x62bd3d31a7b75c098ccf28bc4d4af8c4a191b4b9e451fab4232258079e8b18c4",
      );
      const findings = await runBlock(16818063);
      expect(findings.at(0)).toMatchSnapshot();
    },
    TEST_TIMEOUT,
  );
});
