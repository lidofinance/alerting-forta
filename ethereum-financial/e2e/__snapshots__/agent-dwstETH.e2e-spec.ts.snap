// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`agent-dwstETH e2e tests should process block with high amount of dwstETH minted 1`] = `
Finding {
  "addresses": Array [],
  "alertId": "HIGH-DWSTETH-MINTS-SUM",
  "description": "There were 4357226.75 dwstETH minted during the last 1 hour(s)",
  "labels": Array [],
  "metadata": Object {},
  "name": "ðŸ˜± High amount of dwstETH minted",
  "protocol": "ethereum",
  "severity": 1,
  "type": 2,
}
`;
